server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -268,6 +296,54 @@ class InteractiveSession(
     }
   }
 
+  private def datanucleusJars(livyConf: LivyConf): Seq[String] = {
+    if (sys.env.getOrElse("LIVY_INTEGRATION_TEST", "false").toBoolean) {
+      // datanucleus jars has already been in classpath in integration test
+      Seq.empty
+    } else {
+      val jars = ArrayBuffer.empty[String]
+      val sparkHome = livyConf.sparkHome().get
+      val libdir =
+        if (new File(sparkHome, "RELEASE").isFile) {
+          new File(sparkHome, "lib")
+        } else {
+          new File(sparkHome, "lib_managed/jars")
+        }
+
+      if (libdir.isDirectory()) {
+        jars ++= libdir.listFiles().filter(_.getName.startsWith("datanucleus-"))
+          .map(_.getAbsolutePath)
+      }
+      warn("datanucleus jars can not be found")
+      jars.toSeq
+    }
+  }
+
+  /**
+   * Look for hive-site.xml (for now just ignore spark.files defined in spark-defaults.conf)
+   * 1. First look for hive-site.xml in user request
+   * 2. Then look for that under classpath
+   * 3. Finally look for that under SPARK_HOME/conf
+   * @param livyConf
+   * @return  (hive-site.xml path, whether it is provided by user)
+   */
+  private def hiveSiteFile(requestFiles: List[String],
+                           livyConf: LivyConf): (Option[File], Boolean) = {
+    if (requestFiles.exists(_.split("/").last == "hive-site.xml")) {
+      (None, true)
+    } else {
+      val hiveSiteURL = getClass.getResource("/hive-site.xml")
+      if (hiveSiteURL != null && hiveSiteURL.getProtocol == "file") {
+        return (Some(new File(hiveSiteURL.toURI)), false)
+      }
+      val hiveSiteFile = new File(livyConf.sparkHome().get + "/conf/hive-site.xml")
```

Comment:
This is not needed and actually wrong. If the user has set `SPARK_CONF_DIR`, he explicitly doesn't want Livy to look at `$SPARK_HOME/conf`. Since `SPARK_CONF_DIR` is in the classpath, the previous check suffices.